rust/src/lib.rs

```diff
@@ -2528,23 +2528,37 @@ impl MultiAsset {
         Self(std::collections::BTreeMap::new())
     }
 
+    /// the number of unique policy IDs in the multiasset
     pub fn len(&self) -> usize {
         self.0.len()
     }
 
-    pub fn insert(&mut self, key: &PolicyID, value: &Assets) -> Option<Assets> {
-        self.0.insert(key.clone(), value.clone())
+    /// set (and replace if it exists) all assets with policy {policy_id} to a copy of {assets}
+    pub fn insert(&mut self, policy_id: &PolicyID, assets: &Assets) -> Option<Assets> {
+        self.0.insert(policy_id.clone(), assets.clone())
     }
 
-    pub fn get(&self, key: &PolicyID) -> Option<Assets> {
-        self.0.get(key).map(|v| v.clone())
+    /// all assets under {policy_id}, if any exist, or else None (undefined in JS)
+    pub fn get(&self, policy_id: &PolicyID) -> Option<Assets> {
+        self.0.get(policy_id).map(|v| v.clone())
+    }
+
+    // sets the asset {asset_name} to {value} under policy {policy_id}
+    // returns the previous amount if it was set, or else None (undefined in JS)
+    pub fn set_asset(&mut self, policy_id: &PolicyID, asset_name: &AssetName, value: BigNum) -> Option<BigNum> {
+        self.0.entry(policy_id.clone()).or_default().insert(asset_name, &value)
+    }
+
+    pub fn get_asset(&self, policy_id: &PolicyID, asset_name: &AssetName) -> BigNum {
```

Comment:
There are no comments for `get_asset` also the comments for `set_asset` aren't doc comments (with triple slashes).
Also is it ok to return `BigNum::zero` instead of None in `get_asset` if the asset doesn't exist?

Reply:
Good catch, I pushed a commit to update the comments.

The first time I wrote `get_asset()` I had it returning `Option<BigNum>` but it felt more awkward to use when I was writing the tests so I changed it to just `BigNum`. Does anyone else have any opinion on it? It seems the same thing to me conceptually.